openedx/core/djangoapps/discussions/tests/test_views.py

```diff
@@ -93,3 +127,298 @@ class CourseStaffAuthorizedTest(AuthorizedApiTest):
+        Check we can set LTI configuration
+        """
+        provider_type = 'piazza'
+        for key, value in DATA_LTI_CONFIGURATION.items():
+            payload = {
+                'enabled': True,
+                'provider_type': provider_type,
+                'lti_configuration': {
+                    key: value,
+                }
+            }
+            response = self._post(payload)
+            response = self._get()
+            data = response.json()
+            assert data['enabled']
+            assert data['provider_type'] == provider_type
+            assert data['lti_configuration'][key] == value
+
+    def test_post_lti_invalid(self):
+        """
+        Check validation of LTI configuration ignores unsupported values
+
+        The fields are all open-ended strings and will accept any values.
+        """
+        provider_type = 'piazza'
+        for key, value in DATA_LTI_CONFIGURATION.items():
+            payload = {
+                'enabled': True,
+                'provider_type': provider_type,
+                'lti_configuration': {
+                    key: value,
+                    'ignored-key': 'ignored value',
+                }
+            }
+            response = self._post(payload)
+            assert response
+            response = self._get()
+            data = response.json()
+            assert data['enabled']
+            assert data['provider_type'] == provider_type
+            assert data['lti_configuration'][key] == value
+            assert 'ignored-key' not in data['lti_configuration']
+
+    def test_post_legacy_valid(self):
+        """
+        Check we can set legacy settings configuration
+        """
+        provider_type = 'legacy'
+        for key, value in DATA_LEGACY_CONFIGURATION.items():
+            payload = {
+                'enabled': True,
+                'provider_type': provider_type,
+                'plugin_configuration': {
+                    key: value,
+                }
+            }
+            response = self._post(payload)
+            assert response
+            response = self._get()
+            data = response.json()
+            assert data['enabled']
+            assert data['provider_type'] == provider_type
+            assert data['plugin_configuration'][key] == value
+
+    @ddt.data(
+        {
+            'allow_anonymous': 3,
+        },
+        {
+            'allow_anonymous_to_peers': 3,
+        },
+        {
+            'discussion_blackouts': 3,
+        },
+        {
+            'discussion_topics': 3,
+        },
+    )
+    def test_post_legacy_invalid(self, plugin_configuration):
+        """
+        Check validation of legacy settings configuration
+        """
+        provider_type = 'legacy'
+        payload = {
+            'enabled': True,
+            'provider_type': provider_type,
+            'plugin_configuration': plugin_configuration,
+        }
+        with self.assertRaises(ValidationError):
```

Comment:
Why this test is asserting something which is actually is raised by the test itself? 

I think the following could have worked nicely. 
```
response = self._post(payload)
self.assertTrue(status.is_client_error(response.status_code))
```